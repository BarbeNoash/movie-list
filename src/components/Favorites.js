import React from 'react';

class Favorite extends React.Component {

	constructor(){
		super();

		this.renderFavorite = this.renderFavorite.bind(this);
	}

	renderFavorite(key){
		const movie = this.props.movies[key];

		console.log(movie);
		// return (
		// 	<li key={key}>
		// 		<span>‚≠ê {movie.show_title}</span>
		// 	</li>
		// )
	}

	render(){

		const favoriteMovies = Object.keys(this.props.favorites);

		return (
			<div className="favorite-column">
				<h2>Favorite List</h2>
				<ul>
					{favoriteMovies.map(this.renderFavorite)}
				</ul>
			</div>
		)
	}
}

export default Favorite;