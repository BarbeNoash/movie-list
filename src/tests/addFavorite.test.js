import React from 'react';
import Favorites from '../components/Favorites';
import renderer from 'react-test-renderer';

test('Add Favorite to favorite list', () => {
  const component = renderer.create(
   <li key="movie1">
        <span>‚≠ê "bonjour"</span>
      </li>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.renderFavorite();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});