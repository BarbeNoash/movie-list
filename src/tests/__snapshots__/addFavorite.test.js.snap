// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Favorite to favorite list 1`] = `
<li>
  <span>
    ‚≠ê "bonjour"
  </span>
</li>
`;

